File: LV4/4.3/loworderzeros/loworderzeros.py
limit = 1_000_000
fac = [1]*(limit+1)


def gen_fac():
    for i in range(2, limit+1):
        ans = fac[i-1]*i
        while ans % 10 == 0:
            ans //= 10
        fac[i] = ans % limit


gen_fac()
while True:
    n = int(input())

    if n == 0:
        break

    print(fac[n] % 10)

File: LV4/4.3/loworderzeros/loworderzeros.rb
def gen_fac
  limit = 1_000_000
  fac = [1] * (limit + 1)
  (2..limit + 1).each do |i|
    ans = fac[i - 1] * i
    ans /= 10 while (ans % 10).zero?
    fac[i] = ans % limit
  end
  fac
end

fac = gen_fac

loop do
  n = gets.to_i

  break if n.zero?

  puts fac[n] % 10
end
